core/src/OpenMPTarget/Kokkos_OpenMPTarget_Parallel_MDRange.hpp

```diff
@@ -458,12 +458,7 @@ class ParallelReduce<FunctorType, Kokkos::MDRangePolicy<Traits...>, ReducerType,
   using pointer_type   = typename Analysis::pointer_type;
   using reference_type = typename Analysis::reference_type;
 
-  enum {
-    HasJoin =
-        Impl::FunctorAnalysis<Impl::FunctorPatternInterface::REDUCE, Policy,
-                              FunctorType>::has_join_member_function
-  };
-  enum { UseReducer = is_reducer<ReducerType>::value };
+  static constexpr bool UseReducer = is_reducer<ReducerType>::value;
```

Comment:
This change doesn't look to be part of "Allow using C++23".

Reply:
It's for avoiding warnings building with C++20. This pull request also adds better support for C++20 and thus also tests it.
Once, this pull request actually passes CI, I am happy to discuss splitting it further if that's what people want.

Reply:
Please let me know if you insist on dropping these changes.

Reply:
I'm fine with it.